dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/search/SearchServiceBean.java

```diff
@@ -832,6 +832,24 @@ private String getNonDatasetFieldFacetLabelName(String facetLabelName, String fo
         return facetLabelName;
     }
 
+    private SolrQuery addHighlightFields(SolrQuery solrQuery, Map<String, String> solrFieldsToHightlightOnMap) {
+        Set<String> dynamicDatasetFieldsPrefixes = new HashSet<>();
+
+        for(String field : solrFieldsToHightlightOnMap.keySet()) {
+            if(field.length() >= 8 && SearchDynamicFieldPrefix.contains(field.substring(0, 8))) {
```

Comment:
won't there be a conflict if 2 fields have the same first 8 letters?

Reply:
First 8 letters identify if a field is dynamic or not (dynamic fields starts with 'dsf_xxx_' ). These fields we mask in solor by adding asterisk sign '*' so we don't have to pass them all in a query - thus making the query much shorter.

Non dynamic fields (not starting with 'dsf_xxx_') are passed to the query without any masking.

So fields: dsf_txt_name and dsf_txt_surname will be masked as dsf_txt_* and fields dvName or dvContributorName will be passed as is.

Reply:
Hmm ok, then i propose extracting method to give more meaning like `if(isFieldDynamic)` so then you can even google what's dynamic fields in google.